satpy/tests/reader_tests/test_oli_tirs_l1_tif.py

```diff
@@ -0,0 +1,576 @@
+
+        # Put the datasets to Scene for easy saving
+        scn = Scene()
+        scn["b4"] = ds_b4
+        scn["b4"].attrs["area"] = self.area_def
+        scn["b11"] = ds_b11
+        scn["b11"].attrs["area"] = self.area_def
+        scn["sza"] = ds_sza
+        scn["sza"].attrs["area"] = self.area_def
+
+        # Save the images.  Two images in PNG and two in GeoTIFF
+        scn.save_dataset("b4", writer="geotiff", enhance=False, fill_value=0,
+                         filename=os.path.join(self.base_dir, self.fnames[0]))
+        scn.save_dataset("b11", writer="geotiff", enhance=False, fill_value=0,
+                         filename=os.path.join(self.base_dir, self.fnames[1]))
+        scn.save_dataset("sza", writer="geotiff", enhance=False, fill_value=0,
+                         filename=os.path.join(self.base_dir, self.fnames[3]))
+
+        scn.save_dataset("b4", writer="geotiff", enhance=False, fill_value=0,
+                         filename=self.bad_fname_plat)
+        scn.save_dataset("b4", writer="geotiff", enhance=False, fill_value=0,
+                         filename=self.bad_fname_chan)
+
+        # Write the metadata to a file
+        with open(os.path.join(self.base_dir, self.fnames[2]), "wb") as f:
+            f.write(metadata_text)
+        with open(self.bad_fname_plat2, "wb") as f:
+            f.write(metadata_text)
+
+        self.scn = scn
+
+    def tearDown(self):
+        """Remove the temporary directory created for a test."""
+        try:
+            import shutil
+            shutil.rmtree(self.base_dir, ignore_errors=True)
+        except OSError:
+            pass
+
+    def test_basicload(self):
+        """Test loading a Landsat Scene."""
+        from satpy import Scene
+        scn = Scene(reader="oli_tirs_l1_tif", filenames=[self.fnames[0],
+                                                         self.fnames[1],
+                                                         self.fnames[2]])
+        scn.load(["b04", "b11"])
+
+        # Check dataset is loaded correctly
+        assert scn["b04"].shape == (100, 100)
+        assert scn["b04"].attrs["area"] == self.area_def
+        assert scn["b04"].attrs["saturated"]
+        assert scn["b11"].shape == (100, 100)
+        assert scn["b11"].attrs["area"] == self.area_def
+        with pytest.raises(KeyError, match="saturated"):
+            assert not scn["b11"].attrs["saturated"]
+
+    def test_ch_startend(self):
+        """Test correct retrieval of start/end times."""
+        from datetime import datetime
+
+        from satpy import Scene
+        scn = Scene(reader="oli_tirs_l1_tif", filenames=[self.fnames[0], self.fnames[3], self.fnames[2]])
+        bnds = scn.available_dataset_names()
+        assert bnds == ["b04", "sza"]
+
+        scn.load(["b04"])
+        assert scn.start_time == datetime(2024, 5, 2, 18, 0, 24)
+        assert scn.end_time == datetime(2024, 5, 2, 18, 0, 24)
+
+    def test_loading(self):
```

Comment:
Could you split up this test and others to really have one test case per function?

Reply:
I've split these up now - hopefully a bit better.

Reply:
yes much better, thanks!